File: native/swift/swift-export-standalone-integration-tests/simple/testData/execution/ranges/ranges.kt
// KIND: STANDALONE
// FREE_COMPILER_ARGS: -opt-in=kotlin.native.internal.InternalForKotlinNative
// MODULE: Ranges
// FILE: Ranges.kt

fun distance(some: IntRange): Int = some.endInclusive - some.start

fun materialize(): IntRange = 4 .. 11

fun simple(some: IntRange): IntRange = some.start - 1 .. some.endInclusive + 1

fun nullable(some: IntRange?): IntRange? = some?.let { simple(it) }

//fun total(list: List<IntRange>): IntRange = list.map { it.start }.min() .. list.map { it.endInclusive }.max()

File: native/swift/swift-export-standalone-integration-tests/simple/testData/execution/ranges/ranges.swift
import Ranges
import KotlinRuntime
import Testing

@Test
func testSimple() throws {
    let d = distance(some: 5 ... 10)
    let m = materialize()
    let s = simple(some: 5 ... 10)
    #expect(d == 5)
    #expect(m.contains(8))
    #expect(s.contains(4))
    #expect(s.contains(11))
    #expect(!s.contains(0))
}

@Test
func nullable() throws {
    let n1 = nullable(some: -10 ... -5)
    let n2 = nullable(some: nil)
    #expect(n1 == -11 ... -4)
    #expect(n2 == nil)
}

@Test
func testTotal() throws {
    // Currently crashes with CCE
    //let t = total(list: [1 ... 5, 0 ... 4, 2 ... 7])
    //#expect(t == 0 ... 7)
}
